src/data-loader.ts

```
@@ -923,62 +1065,76 @@ export async function loadSessionData(
-	for (const { file, baseDir } of sortedFilesWithBase) {
-		// Extract session info from file path using its specific base directory
-		const relativePath = path.relative(baseDir, file);
-		const parts = relativePath.split(path.sep);
-
-		// Session ID is the directory name containing the JSONL file
-		const sessionId = parts[parts.length - 2] ?? 'unknown';
-		// Project path is everything before the session ID
-		const joinedPath = parts.slice(0, -2).join(path.sep);
-		const projectPath = joinedPath.length > 0 ? joinedPath : 'Unknown Project';
+	// Set up cache and process files
+	using fileCache = new FileCache<typeof allEntries[number]>('session-cache.json', claudePaths, options);
```

ESLint

[error] Unsafe argument of type error typed assigned to a parameter of type `string[]`.

(ts/no-unsafe-argument)